packages/devtools/lib/src/inspector/diagnostics_node.dart

```diff
@@ -683,38 +683,22 @@ class InspectorSourceLocation {
 
   String get path => JsonUtils.getStringMember(json, 'file');
 
-  String getFile() {
-    final fileName = path;
-    if (fileName == null) {
-      return parent != null ? parent.getFile() : null;
```

Comment:
this part actually shouldn't be removed. it is falling back to the file path specified in the parent for the case of nested location objects (that feature of tracking the locations for parameters that you are looking at testing the performance benefit of removing).

Reply:
Oops, too aggressive put back getFile and useages.